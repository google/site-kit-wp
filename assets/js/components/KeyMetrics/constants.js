/**
 * Key Metrics components - constants.
 *
 * Site Kit by Google, Copyright 2023 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * WordPress dependencies
 */
import { __ } from '@wordpress/i18n';

export const KEY_METRICS_SETUP_CTA_WIDGET_SLUG = 'key-metrics-setup-cta-widget';

export const KEY_METRICS_SELECTION_PANEL_OPENED_KEY =
	'googlesitekit-key-metrics-selection-panel-opened';
export const KEY_METRICS_SELECTION_FORM = 'key-metrics-selection-form';
export const KEY_METRICS_SELECTED = 'key-metrics-selected';
export const EFFECTIVE_SELECTION = 'key-metrics-effective-selection';
export const UNSTAGED_SELECTION = 'key-metrics-unstaged-selection';
export const MIN_SELECTED_METRICS_COUNT = 2;
export const MAX_SELECTED_METRICS_COUNT = 4;
export const MAX_SELECTED_METRICS_COUNT_WITH_CONVERSION_EVENTS = 8;
export const KEY_METRICS_GROUP_CURRENT = {
	SLUG: 'current-selection',
	LABEL: __( 'Current selection', 'google-site-kit' ),
};
export const KEY_METRICS_GROUP_SUGGESTED = {
	SLUG: 'suggested',
	LABEL: __( 'Suggested', 'google-site-kit' ),
};
export const KEY_METRICS_GROUP_VISITORS = {
	SLUG: 'visitors',
	LABEL: __( 'Visitors', 'google-site-kit' ),
};
export const KEY_METRICS_GROUP_DRIVING_TRAFFIC = {
	SLUG: 'driving-traffic',
	LABEL: __( 'Driving traffic', 'google-site-kit' ),
};
export const KEY_METRICS_GROUP_GENERATING_LEADS = {
	SLUG: 'generating-leads',
	LABEL: __( 'Generating leads', 'google-site-kit' ),
};
export const KEY_METRICS_GROUP_SELLING_PRODUCTS = {
	SLUG: 'selling-products',
	LABEL: __( 'Selling products', 'google-site-kit' ),
};
export const KEY_METRICS_GROUP_CONTENT_PERFORMANCE = {
	SLUG: 'content-performance',
	LABEL: __( 'Content performance', 'google-site-kit' ),
};
