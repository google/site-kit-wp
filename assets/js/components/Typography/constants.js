/**
 * Site Kit by Google, Copyright 2025 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

export const TYPE_DISPLAY = 'display';
export const TYPE_HEADLINE = 'headline';
export const TYPE_TITLE = 'title';
export const TYPE_BODY = 'body';
export const TYPE_LABEL = 'label';

export const VALID_TYPES = [
	TYPE_BODY,
	TYPE_DISPLAY,
	TYPE_HEADLINE,
	TYPE_LABEL,
	TYPE_TITLE,
];

export const SIZE_SMALL = 'small';
export const SIZE_MEDIUM = 'medium';
export const SIZE_LARGE = 'large';

export const VALID_SIZES = [ SIZE_SMALL, SIZE_MEDIUM, SIZE_LARGE ];
