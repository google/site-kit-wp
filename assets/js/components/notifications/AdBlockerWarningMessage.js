/**
 * AdBlockerWarningMessage component.
 *
 * Site Kit by Google, Copyright 2024 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * External dependencies
 */
import PropTypes from 'prop-types';

/**
 * WordPress dependencies
 */
import { sprintf, __ } from '@wordpress/i18n';
import { createInterpolateElement } from '@wordpress/element';

/**
 * Internal dependencies
 */
import Link from '../Link';
import WarningNotice from '../WarningNotice';
import ExternalIcon from '../../../svg/icons/external-rounded.svg';

export default function AdBlockerWarningMessage( {
	className = '',
	getHelpLink = '',
	warningMessage = null,
} ) {
	if ( ! warningMessage ) {
		return null;
	}

	return (
		<WarningNotice className={ className }>
			{ createInterpolateElement(
				sprintf(
					/* translators: 1: The warning message. 2: "Get help" text. */
					__( '%1$s. <Link>%2$s</Link>', 'google-site-kit' ),
					warningMessage,
					__( 'Get help', 'google-site-kit' )
				),
				{
					Link: (
						<Link
							href={ getHelpLink }
							external
							hideExternalIndicator
							trailingIcon={
								<ExternalIcon width={ 15 } height={ 15 } />
							}
						/>
					),
				}
			) }
		</WarningNotice>
	);
}

AdBlockerWarningMessage.propTypes = {
	className: PropTypes.string,
	getHelpLink: PropTypes.string,
	warningMessage: PropTypes.string,
};
