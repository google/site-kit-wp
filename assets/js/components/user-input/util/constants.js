/**
 * User Input Answers.
 *
 * Site Kit by Google, Copyright 2021 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * WordPress dependencies
 */
import { __ } from '@wordpress/i18n';

export const USER_INPUT_QUESTIONS_PURPOSE = 'purpose';
export const USER_INPUT_QUESTION_POST_FREQUENCY = 'postFrequency';
export const USER_INPUT_QUESTIONS_GOALS = 'goals';

export const USER_INPUT_QUESTIONS_LIST = [
	USER_INPUT_QUESTIONS_PURPOSE,
	USER_INPUT_QUESTION_POST_FREQUENCY,
	USER_INPUT_QUESTIONS_GOALS,
];

export const USER_INPUT_MAX_ANSWERS = {
	[ USER_INPUT_QUESTIONS_PURPOSE ]: 1,
	[ USER_INPUT_QUESTION_POST_FREQUENCY ]: 1,
	[ USER_INPUT_QUESTIONS_GOALS ]: 3,
};

export const USER_INPUT_CURRENTLY_EDITING_KEY =
	'googlesitekit-user-input-currently-editing';

export const FORM_USER_INPUT_QUESTION_NUMBER = 'user_input_question_number';
export const FORM_USER_INPUT_QUESTION_SNAPSHOT = 'user_input_question_snapshot';
export const USER_INPUT_LEGACY_SITE_PURPOSE_DISMISSED_ITEM_KEY =
	'user-input-legacy-site-purpose-dismissed-item';

/**
 * Gets available questions for user input settings.
 *
 * @since 1.139.0
 * @private
 *
 * @return {Object} Questions object.
 */
export function getUserInputQuestions() {
	return [
		{
			title: __(
				'What is the main purpose of this site?',
				'google-site-kit'
			),
			description: __(
				'Based on your answer, Site Kit will tailor the metrics you see on your dashboard to help you track progress towards your specific goals',
				'google-site-kit'
			),
		},
		{
			title: __(
				'How often do you create new content for this site?',
				'google-site-kit'
			),
			description: __(
				'Based on your answer, Site Kit will suggest new features for your dashboard related to content creation',
				'google-site-kit'
			),
		},
		{
			title: __(
				'What are your top goals for this site?',
				'google-site-kit'
			),
			description: __(
				'Based on your answers, Site Kit will tailor the metrics and recommendations you see on your dashboard to help you make progress in these areas',
				'google-site-kit'
			),
		},
	];
}

/**
 * Gets available answers for user input settings.
 *
 * @since 1.20.0
 * @private
 *
 * @return {Object} Answers object.
 */
export function getUserInputAnswers() {
	return {
		USER_INPUT_ANSWERS_PURPOSE: {
			sell_products_or_service: __(
				'Sell products or services',
				'google-site-kit'
			),
			sell_products: __( 'Sell products', 'google-site-kit' ),
			provide_services: __( 'Provide services', 'google-site-kit' ),
			monetize_content: __(
				'Monetize content (with ads or affiliate links)',
				'google-site-kit'
			),
			publish_blog: __( 'Publish a blog', 'google-site-kit' ),
			publish_news: __( 'Publish news content', 'google-site-kit' ),
			share_portfolio: __(
				'Share a business card or portfolio to represent me or my company online',
				'google-site-kit'
			),
			other: __( 'Other', 'google-site-kit' ),
		},
		USER_INPUT_ANSWERS_POST_FREQUENCY: {
			never: __( 'Never', 'google-site-kit' ),
			daily: __( 'Daily', 'google-site-kit' ),
			weekly: __( 'Weekly', 'google-site-kit' ),
			monthly: __( 'Monthly', 'google-site-kit' ),
			other: __( 'Other', 'google-site-kit' ),
		},
		USER_INPUT_ANSWERS_GOALS: {
			retaining_visitors: __(
				'Retain visitors, turn them into loyal readers or customers',
				'google-site-kit'
			),
			improving_performance: __(
				'Improve speed and performance',
				'google-site-kit'
			),
			finding_new_topics: __(
				'Find new topics to write about that connect with my audience',
				'google-site-kit'
			),
			growing_audience: __( 'Grow my audience', 'google-site-kit' ),
			expanding_business: __(
				'Expand my business into new cities, states or markets',
				'google-site-kit'
			),
			generating_revenue: __(
				'Generate more revenue',
				'google-site-kit'
			),
			help_better_rank: __(
				'Help my content rank in a better position in Google search results',
				'google-site-kit'
			),
			understanding_content_performance: __(
				'Understand which content is performing best',
				'google-site-kit'
			),
			encourage_to_post: __(
				'Tips for generating and posting engaging content updates',
				'google-site-kit'
			),
			other: __( 'Other', 'google-site-kit' ),
		},
	};
}

/**
 * Gets available answer descriptions for user input settings.
 *
 * @since 1.139.0
 * @private
 *
 * @return {Object} Answer descriptions object.
 */
export function getUserInputAnswersDescription() {
	return {
		USER_INPUT_ANSWERS_PURPOSE: {
			sell_products_or_service: __(
				'E.g. selling products like devices, apparel, equipment, etc. or offering services like courses, consulting, tutoring, etc.',
				'google-site-kit'
			),
			sell_products: __(
				'E.g. selling devices, apparel, equipment, etc.',
				'google-site-kit'
			),
			provide_services: __(
				'E.g. offering courses, consulting, tutoring, etc.',
				'google-site-kit'
			),
			monetize_content: __(
				'Using display ads, affiliate links, sponsored content, etc.',
				'google-site-kit'
			),
			publish_blog: __(
				'Writing on a topic youâ€™re passionate about, no focus on monetizing content',
				'google-site-kit'
			),
			publish_news: __(
				'E.g. local news, investigative pieces, interviews, etc.',
				'google-site-kit'
			),
			share_portfolio: __(
				'My website represents me or my company',
				'google-site-kit'
			),
			other: undefined,
		},
	};
}
