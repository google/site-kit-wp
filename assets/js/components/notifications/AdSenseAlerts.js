/**
 * AdSenseAlerts component.
 *
 * Site Kit by Google, Copyright 2021 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * WordPress dependencies
 */
import { Fragment } from '@wordpress/element';
import { __, _x } from '@wordpress/i18n';

/**
 * Internal dependencies
 */
import { useSelect } from 'googlesitekit-data';
import { CORE_MODULES } from '../../googlesitekit/modules/datastore/constants';
import { MODULES_ADSENSE } from '../../modules/adsense/datastore/constants';
import NotificationAlertSVG from '../../../svg/graphics/notification-alert.svg';
import BannerNotification from '../notifications/BannerNotification';

function AdSenseAlerts() {
	const adSenseModuleConnected = useSelect( ( select ) =>
		select( CORE_MODULES ).isModuleConnected( 'adsense' )
	);
	const accountID = useSelect( ( select ) =>
		select( MODULES_ADSENSE ).getAccountID()
	);
	const notifications = useSelect( ( select ) =>
		select( MODULES_ADSENSE ).getNotifications()
	);

	if (
		! adSenseModuleConnected ||
		! accountID ||
		notifications === undefined
	) {
		return null;
	}

	return (
		<Fragment>
			{ notifications.map(
				( {
					id,
					title,
					description,
					format,
					ctaURL,
					ctaLabel,
					ctaTarget,
					severity,
					isDismissable,
				} ) => (
					<BannerNotification
						className="googlesitekit-adsense-alert"
						key={ id }
						id={ id }
						title={ title || '' }
						description={ description }
						WinImageSVG={ NotificationAlertSVG }
						format={ format || 'small' }
						ctaLink={ ctaURL }
						ctaLabel={ ctaLabel }
						ctaTarget={ ctaTarget }
						type={ severity }
						dismiss={ __( 'OK, Got it!', 'google-site-kit' ) }
						isDismissible={ isDismissable || true }
						logo
						module="adsense"
						moduleName={ _x(
							'AdSense',
							'Service name',
							'google-site-kit'
						) }
						dismissExpires={ 0 }
						showOnce={ false }
					/>
				)
			) }
		</Fragment>
	);
}

export default AdSenseAlerts;
