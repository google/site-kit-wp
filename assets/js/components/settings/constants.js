/**
 * Settings constants.
 *
 * Site Kit by Google, Copyright 2024 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { MODULE_SLUG_ADS } from '../../modules/ads/datastore/constants';
import { MODULE_SLUG_SIGN_IN_WITH_GOOGLE } from '../../modules/sign-in-with-google/datastore/constants';
import { MODULE_SLUG_READER_REVENUE_MANAGER } from '../../modules/reader-revenue-manager/datastore/constants';

export const NEW_MODULES = [
	MODULE_SLUG_ADS,
	MODULE_SLUG_READER_REVENUE_MANAGER,
	MODULE_SLUG_SIGN_IN_WITH_GOOGLE,
];

export const BETA_MODULES = [ MODULE_SLUG_SIGN_IN_WITH_GOOGLE ];

export const EXPERIMENTAL_MODULES = [];
