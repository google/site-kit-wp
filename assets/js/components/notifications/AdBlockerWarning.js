/**
 * AdSense AdBlockerWarning component.
 *
 * Site Kit by Google, Copyright 2021 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * External dependencies
 */
import PropTypes from 'prop-types';

/**
 * Internal dependencies
 */
import { useSelect } from 'googlesitekit-data';
import { CORE_SITE } from '../../googlesitekit/datastore/site/constants';
import { CORE_MODULES } from '../../googlesitekit/modules/datastore/constants';
import AdBlockerWarningMessage from './AdBlockerWarningMessage';

export default function AdBlockerWarning( { moduleSlug, className } ) {
	const storeName = useSelect( ( select ) =>
		select( CORE_MODULES ).getModuleStoreName( moduleSlug )
	);
	const adBlockerWarningMessage = useSelect( ( select ) =>
		select( storeName )?.getAdBlockerWarningMessage()
	);
	const getHelpLink = useSelect( ( select ) =>
		select( CORE_SITE ).getDocumentationLinkURL(
			`${ moduleSlug }-ad-blocker-detected`
		)
	);

	return (
		<AdBlockerWarningMessage
			className={ className }
			getHelpLink={ getHelpLink }
			warningMessage={ adBlockerWarningMessage }
		/>
	);
}

AdBlockerWarning.propTypes = {
	className: PropTypes.string,
	moduleSlug: PropTypes.string.isRequired,
};
