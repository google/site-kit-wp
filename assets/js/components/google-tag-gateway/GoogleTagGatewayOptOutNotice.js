/**
 * Google Tag Gateway Opt-Out Notice component.
 *
 * Site Kit by Google, Copyright 2025 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * WordPress dependencies
 */
import { createInterpolateElement, useCallback } from '@wordpress/element';
import { __ } from '@wordpress/i18n';

/**
 * Internal dependencies
 */
import { useSelect, useDispatch } from 'googlesitekit-data';
import { CORE_SITE } from '@/js/googlesitekit/datastore/site/constants';
import { CORE_USER } from '@/js/googlesitekit/datastore/user/constants';
import { CORE_NOTIFICATIONS } from '@/js/googlesitekit/notifications/datastore/constants';
import { GTG_AUTO_ENABLE_NOTIFICATION } from '@/js/googlesitekit/notifications/constants';
import Notice from '@/js/components/Notice';
import { trackEvent } from '@/js/util';
import useViewContext from '@/js/hooks/useViewContext';
import withIntersectionObserver from '@/js/util/withIntersectionObserver';

const GTG_OPT_OUT_NOTICE_DISMISSED_ITEM_KEY = 'gtg-opt-out-notice-dismissed';

const NoticeWithIntersectionObserver = withIntersectionObserver( Notice );

export default function GoogleTagGatewayOptOutNotice() {
	const viewContext = useViewContext();

	const { setIsGTGDefault, saveGoogleTagGatewaySettings } =
		useDispatch( CORE_SITE );
	const { dismissNotification } = useDispatch( CORE_NOTIFICATIONS );
	const { dismissItem } = useDispatch( CORE_USER );

	const isGTGModuleConnected = useSelect( ( select ) =>
		select( CORE_SITE ).isAnyGoogleTagGatewayModuleConnected()
	);

	const isGoogleTagGatewayEnabled = useSelect( ( select ) =>
		select( CORE_SITE ).isGoogleTagGatewayEnabled()
	);

	const isGTGDefault = useSelect( ( select ) =>
		select( CORE_SITE ).isGTGDefault()
	);

	const isGTGHealthy = useSelect( ( select ) =>
		select( CORE_SITE ).isGTGHealthy()
	);

	const isScriptAccessEnabled = useSelect( ( select ) =>
		select( CORE_SITE ).isScriptAccessEnabled()
	);

	const learnMoreURL = useSelect( ( select ) => {
		return select( CORE_SITE ).getDocumentationLinkURL(
			'google-tag-gateway-introduction'
		);
	} );

	const isNoticeDismissed = useSelect( ( select ) =>
		select( CORE_USER ).isItemDismissed(
			GTG_OPT_OUT_NOTICE_DISMISSED_ITEM_KEY
		)
	);

	const isAutoEnableNotificationVisible = useSelect(
		( select ) =>
			! select( CORE_NOTIFICATIONS ).isNotificationDismissed(
				GTG_AUTO_ENABLE_NOTIFICATION
			)
	);

	const handleGotItClick = useCallback( async () => {
		trackEvent( `${ viewContext }_gtg-opt-out-notice`, 'dismiss_notice' );

		await dismissItem( GTG_OPT_OUT_NOTICE_DISMISSED_ITEM_KEY );

		if ( isAutoEnableNotificationVisible ) {
			dismissNotification( GTG_AUTO_ENABLE_NOTIFICATION );
		}
	}, [
		viewContext,
		dismissItem,
		isAutoEnableNotificationVisible,
		dismissNotification,
	] );

	const handleOptOutClick = useCallback( async () => {
		trackEvent( `${ viewContext }_gtg-opt-out-notice`, 'click_opt_out' );

		// Mark settings as no longer default (user has explicitly opted out)
		setIsGTGDefault( false );
		await saveGoogleTagGatewaySettings();

		// Dismiss this notice persistently
		await dismissItem( GTG_OPT_OUT_NOTICE_DISMISSED_ITEM_KEY );

		// If auto-enable notification is still visible, dismiss it too
		if ( isAutoEnableNotificationVisible ) {
			dismissNotification( GTG_AUTO_ENABLE_NOTIFICATION );
		}
	}, [
		viewContext,
		setIsGTGDefault,
		saveGoogleTagGatewaySettings,
		dismissItem,
		isAutoEnableNotificationVisible,
		dismissNotification,
	] );

	if (
		! isGTGModuleConnected ||
		isGoogleTagGatewayEnabled ||
		! isGTGDefault ||
		isNoticeDismissed ||
		! isGTGHealthy ||
		! isScriptAccessEnabled
	) {
		return null;
	}

	const description = createInterpolateElement(
		__(
			'Starting in October 2025, Google tag gateway for advertisers will gradually be enabled. You can opt out now before the change happens and it wonâ€™t be automatically enabled for you.',
			'google-site-kit'
		),
		{
			a: (
				<a
					href={ learnMoreURL }
					target="_blank"
					rel="noopener noreferrer"
					onClick={ () => {
						trackEvent(
							`${ viewContext }_gtg-opt-out-notice`,
							'click_learn_more_link'
						);
					} }
				>
					{ __( 'Learn more', 'google-site-kit' ) }
				</a>
			),
		}
	);

	return (
		<NoticeWithIntersectionObserver
			type={ Notice.TYPES.NEW }
			description={ description }
			className="googlesitekit-gtg-settings-notice"
			ctaButton={ {
				label: __( 'Got it', 'google-site-kit' ),
				onClick: handleGotItClick,
			} }
			dismissButton={ {
				label: __( 'Opt out', 'google-site-kit' ),
				onClick: handleOptOutClick,
			} }
			onInView={ () => {
				trackEvent(
					`${ viewContext }_gtg-opt-out-notice`,
					'view_notice'
				);
			} }
		/>
	);
}
