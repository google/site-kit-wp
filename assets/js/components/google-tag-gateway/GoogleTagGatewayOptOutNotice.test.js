/**
 * GoogleTagGatewayOptOutNotice component tests.
 *
 * Site Kit by Google, Copyright 2025 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * External dependencies
 */
import { useIntersection as mockUseIntersection } from 'react-use';

/**
 * Internal dependencies
 */
import {
	createTestRegistry,
	provideModules,
	provideSiteInfo,
	provideUserInfo,
	render,
	fireEvent,
	waitFor,
} from '../../../../tests/js/test-utils';
import { CORE_SITE } from '@/js/googlesitekit/datastore/site/constants';
import { CORE_USER } from '@/js/googlesitekit/datastore/user/constants';
import { CORE_NOTIFICATIONS } from '@/js/googlesitekit/notifications/datastore/constants';
import { GTG_AUTO_ENABLE_NOTIFICATION } from '@/js/googlesitekit/notifications/constants';
import { DEFAULT_NOTIFICATIONS } from '@/js/googlesitekit/notifications/register-defaults';
import { MODULE_SLUG_ANALYTICS_4 } from '@/js/modules/analytics-4/constants';
import { MODULE_SLUG_ADS } from '@/js/modules/ads/constants';
import { MODULE_SLUG_TAGMANAGER } from '@/js/modules/tagmanager/constants';
import * as tracking from '@/js/util/tracking';
import GoogleTagGatewayOptOutNotice, {
	GTG_OPT_OUT_NOTICE_DISMISSED_ITEM_KEY,
} from './GoogleTagGatewayOptOutNotice';
import { VIEW_CONTEXT_SETTINGS } from '@/js/googlesitekit/constants';

jest.mock( 'react-use', () => ( {
	...jest.requireActual( 'react-use' ),
	useIntersection: jest.fn(),
} ) );

const mockTrackEvent = jest.spyOn( tracking, 'trackEvent' );
mockTrackEvent.mockImplementation( () => Promise.resolve() );

describe( 'GoogleTagGatewayOptOutNotice', () => {
	let registry;

	const gtgSettings = {
		isEnabled: false,
		isGTGHealthy: true,
		isScriptAccessEnabled: true,
		isGTGDefault: true,
	};

	beforeEach( () => {
		mockTrackEvent.mockClear();
		mockUseIntersection.mockClear();

		mockUseIntersection.mockImplementation( () => ( {
			isIntersecting: false,
			intersectionRatio: 0,
		} ) );

		registry = createTestRegistry();

		provideUserInfo( registry );
		provideSiteInfo( registry );
		provideModules( registry, [
			{
				slug: MODULE_SLUG_ANALYTICS_4,
				active: true,
				connected: true,
			},
			{
				slug: MODULE_SLUG_ADS,
				active: true,
				connected: false,
			},
			{
				slug: MODULE_SLUG_TAGMANAGER,
				active: true,
				connected: false,
			},
		] );

		registry
			.dispatch( CORE_SITE )
			.receiveGetGoogleTagGatewaySettings( gtgSettings );

		registry.dispatch( CORE_USER ).receiveGetDismissedItems( [] );
	} );

	afterEach( () => {
		jest.clearAllMocks();
	} );

	describe( 'visibility conditions', () => {
		it( 'should render when all conditions are met', () => {
			const { container, getByText } = render(
				<GoogleTagGatewayOptOutNotice />,
				{
					registry,
				}
			);

			expect( container ).toMatchSnapshot();

			expect(
				getByText(
					/Starting in October 2025, Google tag gateway for advertisers will gradually be enabled/
				)
			).toBeInTheDocument();
		} );

		it( 'should not render when no GTG modules are connected', () => {
			provideModules( registry, [
				{
					slug: MODULE_SLUG_ANALYTICS_4,
					active: true,
					connected: false,
				},
				{
					slug: MODULE_SLUG_ADS,
					active: true,
					connected: false,
				},
				{
					slug: MODULE_SLUG_TAGMANAGER,
					active: true,
					connected: false,
				},
			] );

			const { container } = render( <GoogleTagGatewayOptOutNotice />, {
				registry,
			} );

			expect( container ).toBeEmptyDOMElement();
		} );

		it( 'should not render when GTG is already enabled', () => {
			registry.dispatch( CORE_SITE ).receiveGetGoogleTagGatewaySettings( {
				...gtgSettings,
				isEnabled: true,
			} );

			const { container } = render( <GoogleTagGatewayOptOutNotice />, {
				registry,
			} );

			expect( container ).toBeEmptyDOMElement();
		} );

		it( 'should not render when isGTGDefault is false', () => {
			registry.dispatch( CORE_SITE ).receiveGetGoogleTagGatewaySettings( {
				...gtgSettings,
				isGTGDefault: false,
			} );

			const { container } = render( <GoogleTagGatewayOptOutNotice />, {
				registry,
			} );

			expect( container ).toBeEmptyDOMElement();
		} );

		it( 'should not render when notice is already dismissed', () => {
			registry
				.dispatch( CORE_USER )
				.receiveGetDismissedItems( [
					GTG_OPT_OUT_NOTICE_DISMISSED_ITEM_KEY,
				] );

			const { container } = render( <GoogleTagGatewayOptOutNotice />, {
				registry,
			} );

			expect( container ).toBeEmptyDOMElement();
		} );

		it( 'should not render when GTG is not healthy', () => {
			registry.dispatch( CORE_SITE ).receiveGetGoogleTagGatewaySettings( {
				...gtgSettings,
				isGTGHealthy: false,
			} );

			const { container } = render( <GoogleTagGatewayOptOutNotice />, {
				registry,
			} );

			expect( container ).toBeEmptyDOMElement();
		} );

		it( 'should not render when script access is not enabled', () => {
			registry.dispatch( CORE_SITE ).receiveGetGoogleTagGatewaySettings( {
				...gtgSettings,
				isScriptAccessEnabled: false,
			} );

			const { container } = render( <GoogleTagGatewayOptOutNotice />, {
				registry,
			} );

			expect( container ).toBeEmptyDOMElement();
		} );
	} );

	it( 'should render with correct description and "Opt out" CTA', () => {
		const { getByText, getByRole } = render(
			<GoogleTagGatewayOptOutNotice />,
			{
				registry,
			}
		);

		expect(
			getByText(
				/Starting in October 2025, Google tag gateway for advertisers will gradually be enabled/
			)
		).toBeInTheDocument();

		expect(
			getByRole( 'button', { name: 'Opt out' } )
		).toBeInTheDocument();
	} );

	describe( 'interactions', () => {
		it( 'should track view event when component comes into view', () => {
			mockUseIntersection.mockImplementation( () => ( {
				isIntersecting: true,
				intersectionRatio: 0.5,
			} ) );

			render( <GoogleTagGatewayOptOutNotice />, {
				registry,
				viewContext: VIEW_CONTEXT_SETTINGS,
			} );

			expect( mockTrackEvent ).toHaveBeenCalledWith(
				'settings_gtg-opt-out-notice',
				'view_notice'
			);
		} );

		it( 'should track event, set isGTGDefault to false, and dismiss notice when "Opt out" button is clicked', async () => {
			const notification =
				DEFAULT_NOTIFICATIONS[ GTG_AUTO_ENABLE_NOTIFICATION ];

			registry
				.dispatch( CORE_NOTIFICATIONS )
				.registerNotification(
					GTG_AUTO_ENABLE_NOTIFICATION,
					notification
				);

			const { getByRole } = render( <GoogleTagGatewayOptOutNotice />, {
				registry,
				viewContext: VIEW_CONTEXT_SETTINGS,
			} );

			const optOutButton = getByRole( 'button', { name: 'Opt out' } );
			fireEvent.click( optOutButton );

			await waitFor( () => {
				expect( mockTrackEvent ).toHaveBeenCalledWith(
					'settings_gtg-opt-out-notice',
					'click_opt_out'
				);
			} );

			// Verify isGTGDefault is set to false.
			expect( registry.select( CORE_SITE ).isGTGDefault() ).toBe( false );
		} );
	} );
} );
