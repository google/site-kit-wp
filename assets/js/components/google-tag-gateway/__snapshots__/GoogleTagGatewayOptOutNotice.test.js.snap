// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GoogleTagGatewayOptOutNotice visibility conditions should render when all conditions are met 1`] = `
<div>
  <div
    class="googlesitekit-notice googlesitekit-notice--new"
  >
    <div
      class="googlesitekit-notice__icon"
    >
      <svg />
    </div>
    <div
      class="googlesitekit-notice__content"
    >
      <p
        class="googlesitekit-notice__description"
      >
        Starting in October 2025, Google tag gateway for advertisers will gradually be enabled. 
        <br />
        You can opt out now before the change happens and it wonâ€™t be automatically enabled for you.
      </p>
    </div>
    <div
      class="googlesitekit-notice__action"
    >
      <button
        class="mdc-button mdc-button--tertiary"
        target="_self"
      >
        <span
          class="mdc-button__label"
        >
          Opt out
        </span>
      </button>
    </div>
  </div>
</div>
`;
